index/scorch/persister.go

```diff
@@ -35,16 +36,20 @@ import (
 
 var DefaultChunkFactor uint32 = 1024
 
-// Arbitrary number, need to make it configurable.
-// Lower values like 10/making persister really slow
-// doesn't work well as it is creating more files to
-// persist for in next persist iteration and spikes the # FDs.
-// Ideal value should let persister also proceed at
-// an optimum pace so that the merger can skip
-// many intermediate snapshots.
-// This needs to be based on empirical data.
-// TODO - may need to revisit this approach/value.
-var epochDistance = uint64(5)
+var DefaultWorkLoopWait int = 2000 // ms
+
+var DefaultPersisterNapUnderNumFiles int = 1000
+
+type persisterOptions struct {
+	// WorkLoopWait controls the wait/delay injected into
+	// persistence workloop to improve the chances for
+	// a healthier and heavier in-memory merging
+	WorkLoopWait int
+
+	// PersisterNapUnderNumFiles threshold mandates the persister to wait
+	// until the merger compacts the files on disk within this threshold
```

Comment:
this doc comment looks wrong, as is more something like, when PersisterNapTimeMSec > 0, and the number of files is less than PersisterNapUnderNumFiles, then the persister will sleep PersisterNapTimeMSec amount of time, which can help blah blah blah